spec.html

```diff
@@ -8553,7 +8553,7 @@ <h1>FunctionDeclarationInstantiation ( _func_, _argumentsList_ )</h1>
   <emu-clause id="sec-built-in-function-objects">
     <h1>Built-in Function Objects</h1>
     <p>The built-in function objects defined in this specification may be implemented as either ECMAScript function objects (<emu-xref href="#sec-ecmascript-function-objects"></emu-xref>) whose behaviour is provided using ECMAScript code or as implementation provided function exotic objects whose behaviour is provided in some other manner. In either case, the effect of calling such functions must conform to their specifications. An implementation may also provide additional built-in function objects that are not defined in this specification.</p>
-    <p>If a built-in function object is implemented as an exotic object it must have the ordinary object behaviour specified in <emu-xref href="#sec-ordinary-object-internal-methods-and-internal-slots"></emu-xref>. All such function exotic objects also have [[Prototype]], [[Extensible]], [[Realm]], and [[ScriptOrModule]] internal slots.</p>
```

Comment:
it seems very strange to me that all functions wouldn't all have the same slots, even if it's null for this subset. What's the benefit of dropping it?

Reply:
exotic objects don't all have the same slots

Reply:
Surely not, but generally each category of thing shares the same slots, even if the values are different.

Reply:
You might say all functions have `[[Prototype]], [[Extensible]], and [[Realm]]`, to satisfy the invariants of the language, and [[ScriptOrModule]] is an additional slot on ECMAScript Source Text functions (and not the only one) as they come from a script or module.

Reply:
> it seems very strange to me that all functions wouldn't all have the same slots, even if it's null for this subset.

[Table 28](https://tc39.es/ecma262/#table-internal-slots-of-ecmascript-function-objects) lists 11 internal slots (in addition to `[[Prototype]]` and `[[Extensible]]`) that every ordinary function must have. An exotic function isn't required to have *any* of those, except (e.g.) where it's implementing a built-in, in which case it's currently required to have 4 of those 13 slots. This PR just reduces that 4 to 3. Are you saying that *every* function object should be required to have all 13?

Reply:
> You might say all functions have `[[Prototype]], [[Extensible]], and [[Realm]]`, to satisfy the invariants of the language,

That would be incorrect.
- Bound function exotic objects don't have `[[Realm]]`.
- Proxy function objects don't have any of those (just `[[ProxyHandler]]` and `[[ProxyTarget]]`).
- Non-standard exotic functions aren't required to have any internal slots.

Reply:
Right sorry I was more trying to say, whatever all objects happen to have is what they share, it's a cyclical definition.